### ISSUE TITLE
dashes in existing frontmatter in source HTML files become escaped.

### ISSUE THREAD

**Issue by schneidr** [JohannesKaufmann/html-to-markdown#26]
**Describe the bug**

dashes in existing frontmatter in source HTML files become escaped.

**HTML Input**
```yaml
---
type: page
layout: reisebericht
title: Hamar
date: '2018-10-24 22:32:03 +0100'
weight: 2
tags:
- Norwegen
- Hamar
url: /2018/10-norwegen/02-hamar/
description: Ein Spaziergang durch Hamar, Shoppen und ein Besuch im Norsk jernbanemusem, dem norwegischen Eisenbahnmuseum.
image: files/2018/10-Norwegen/Hamar_Titel.jpg
---
```

**Generated Markdown**
````yaml
\-\-\-
type: page
layout: reisebericht
title: Hamar
date: '2018-10-24 22:32:03 +0100'
weight: 2
tags:
\- Norwegen
\- Hamar
url: /2018/10-norwegen/02-hamar/
description: Ein Spaziergang durch Hamar, Shoppen und ein Besuch im Norsk jernbanemusem, dem norwegischen Eisenbahnmuseum.
image: files/2018/10-Norwegen/Hamar\_Titel.jpg
\-\-\-
````

**Expected Markdown**
````yaml
---
type: page
layout: reisebericht
title: Hamar
date: '2018-10-24 22:32:03 +0100'
weight: 2
tags:
- Norwegen
- Hamar
url: /2018/10-norwegen/02-hamar/
description: Ein Spaziergang durch Hamar, Shoppen und ein Besuch im Norsk jernbanemusem, dem norwegischen Eisenbahnmuseum.
image: files/2018/10-Norwegen/Hamar_Titel.jpg
---
````

**Additional context**
This problem occurs when the source files already contain frontmatter (for example when converting Hugo .html files to .md).

**Comment by JohannesKaufmann** [comment]
@schneidr The escaping works as intended. Only actual html elements should generate a markdown identifier (like a dash). 

I’m not sure if frontmatter support should be part of the core library, but it **could be added as a plugin/helper function**.

----

One idea is to use [github.com/gernest/front](https://github.com/gernest/front), but that would parse the frontmatter into a map (thus in random order) and you would still have to stringify the data structure afterwards.


```go
html := `---
type: page
tags:
- Berlin
---

<h1>Title</h1>
`

// 1. Split into Frontmatter and Body
m := front.NewMatter()
m.Handle("---", front.YAMLHandler)
f, body, err := m.Parse(strings.NewReader(html))
if err != nil {
	log.Fatal(err)
}

fmt.Printf("The front matter is:\n%#v\n", f)
fmt.Printf("The body is:\n%q\n", body)

// 2. Convert the HTML
conv := md.NewConverter("", true, nil)

markdown, err := conv.ConvertString(body)
if err != nil {
	log.Fatal(err)
}

fmt.Println(markdown)
```


----


The best option would probably be to
- remove the content between the frontmatter delimiters
- convert the HTML into markdown 
- prepend the frontmatter to the markdown

That way the frontmatter stays in the correct formatting.

**Comment by JohannesKaufmann** [comment]
TLDR: I will think about how I could add this functionality to the library as a helper function.

**Comment by JohannesKaufmann** [comment]
I added a new `MoveFrontMatter` Plugin in the latest commit 🚀

```go
conv.Use(plugin.EXPERIMENTALMoveFrontMatter())

// or with the allowed delimiter characters
conv.Use(plugin.EXPERIMENTALMoveFrontMatter('-', '+'))
```

---

@schneidr please try to **break it** and report any bugs 😉


If you **don't** find any bugs, please let me know. Then I will remove the `EXPERIMENTAL`-Prefix and release a new version.
